src/index.ts

```diff
@@ -776,20 +780,64 @@ export default runExtension({
                   ).includes("<%NOCURSOR%>"),
                   triggerUid: parentUid,
                 };
-                if (keepButton) {
-                  createBlock({
-                    node: { text: "" },
-                    parentUid,
-                  }).then((targetUid) =>
-                    sbBomb({
-                      ...props,
-                      target: {
-                        uid: targetUid,
-                        start: 0,
-                        end: 0,
-                      },
-                    }).then((n) => n === 0 && deleteBlock(targetUid))
-                  );
+                                  
+                if (applyToSibling) {
+                  const sbParentTree = getShallowTreeByParentUid(getParentUidByBlockUid(parentUid));
+                  const siblingIndex = sbParentTree.findIndex(obj => obj.uid === parentUid) + (applyToSibling === "previous" ? -1 : 1)
+                  const siblingUid = sbParentTree[siblingIndex]?.uid;
+                  const siblingText = getTextByBlockUid(siblingUid);
+
+                  updateBlock({
+                    uid: parentUid,
+                    text: clearBlock && keepButton
+                      ? full
+                      : clearBlock
+                      ? ""
+                      : `${text.substring(0, index)}${text.substring(
+                          index + full.length
+                        )}`,
+                  });
+                  updateBlock({
+                      uid: siblingUid,
```

Comment:
we should handle the case where `siblingUid` doesn't exist and we need to create a block instead